crates/edr_evm/src/chain_spec.rs

```diff
@@ -1,55 +1,140 @@
 
-use crate::{transaction::remote::EthRpcTransaction, EthRpcBlock, IntoRemoteBlock};
+use crate::{
+    hardfork::{self, Activations},
+    transaction::remote::EthRpcTransaction,
+    BlockReceipt, EthBlockData, EthRpcBlock, RemoteBlockConversionError,
+};
 
 /// A trait for defining a chain's associated types.
 // Bug: https://github.com/rust-lang/rust-clippy/issues/12927
 #[allow(clippy::trait_duplication_in_bounds)]
 pub trait ChainSpec:
-    Debug
-    + alloy_rlp::Encodable
-    + revm::primitives::ChainSpec<
-        Transaction: alloy_rlp::Encodable + Clone + Debug + PartialEq + Eq + SignedTransaction,
+    alloy_rlp::Encodable
+    + EthHeaderConstants
+    + revm::ChainSpec<
+        Block: BlockEnvConstructor<Self>,
+        Transaction: alloy_rlp::Encodable
+                         + Clone
+                         + Debug
+                         + PartialEq
+                         + Eq
+                         + SignedTransaction
+                         + TryFrom<
+            <Self as RpcSpec>::RpcTransaction,
+            Error = Self::RpcTransactionConversionError,
+        >,
     > + RpcSpec<
-        RpcBlock<<Self as RpcSpec>::RpcTransaction>: EthRpcBlock + IntoRemoteBlock<Self>,
+        ExecutionReceipt<FilterLog>: Debug,
+        RpcBlock<<Self as RpcSpec>::RpcTransaction>: EthRpcBlock
+                                                         + TryInto<
+            EthBlockData<Self>,
+            Error = Self::RpcBlockConversionError,
+        >,
+        RpcReceipt: Debug + TryInto<BlockReceipt<Self>, Error = Self::RpcReceiptConversionError>,
         RpcTransaction: EthRpcTransaction,
-    > + RpcSpec<RpcBlock<B256>: EthRpcBlock>
+    > + RpcSpec<
+        ExecutionReceipt<ExecutionLog>: alloy_rlp::Encodable
+                                            + MapReceiptLogs<
+            ExecutionLog,
+            FilterLog,
+            Self::ExecutionReceipt<FilterLog>,
+        >,
+        RpcBlock<B256>: EthRpcBlock,
+    >
 {
-}
+    /// Type representing a builder that constructs an execution receipt.
+    type ReceiptBuilder: ExecutionReceiptBuilder<
+        Self,
+        Receipt = Self::ExecutionReceipt<ExecutionLog>,
+    >;
 
-/// A supertrait for [`ChainSpec`] that is safe to send between threads.
-pub trait SyncChainSpec: ChainSpec<Transaction: Send + Sync> + Send + Sync + 'static {}
+    /// Type representing an error that occurs when converting an RPC block.
+    type RpcBlockConversionError: Debug + std::error::Error;
 
-impl<ChainSpecT> SyncChainSpec for ChainSpecT where
-    ChainSpecT: ChainSpec<Transaction: Send + Sync> + Send + Sync + 'static
-{
-}
+    /// Type representing an error that occurs when converting an RPC receipt.
+    type RpcReceiptConversionError: Debug + std::error::Error;
+
+    /// Type representing an error that occurs when converting an RPC
+    /// transaction.
+    type RpcTransactionConversionError: Debug + std::error::Error;
```

Comment:
`std::error::Error` already requires `Debug`, so I think the explicit `Debug` bound could be removed.

Reply:
Addressed by https://github.com/NomicFoundation/edr/pull/534/commits/cbdb02b85bf632a5d49e379fa5407a8dd39e55e3